### PR TITLE
Ledger: Move FeeManager / UTXOCache instantiation in the Ledger

### PR DESCRIPTION
**Pull request by Geod24**
```
The instantiation happening in the node is a leftover of the time
where dependency injection was done at the class level.
Now that we just pass a ManagedDatabase, there is no need for it.
If tests want to implement a specific behavior for one of those
classes, they can re-introduce the dependency injection in the Ledger.
```

### CODE REVIEW

**Comment by Geod24** [comment]
Rebased

File: source/agora/consensus/Ledger.d [hunk 1]
```diff
@@ -91,21 +91,19 @@ public class NodeLedger : Ledger
...
                               when a block was added to the ledger
 
     ***************************************************************************/
 
     public this (immutable(ConsensusParams) params,
-        Engine engine, UTXOSet utxo_set, IBlockStorage storage,
-        EnrollmentManager enroll_man, TransactionPool pool,
-        FeeManager fee_man,
+        ManagedDatabase database, IBlockStorage storage,
+        Engine engine, EnrollmentManager enroll_man, TransactionPool pool,
```
Comment (omerfirmak) [inline source/agora/consensus/Ledger.d:106]: We could do the same for `engine` as well, it is not stateful and pretty lightweight too. There is no reason to share it.

**Review by omerfirmak — APPROVED** [review]

File: source/agora/consensus/Ledger.d (hunk 1 above)
Comment (omerfirmak) [reply source/agora/consensus/Ledger.d:106]: Engine could even be disposible, created only in the scope that it is needed.

Comment (Geod24) [reply source/agora/consensus/Ledger.d:106]: Yeah I wanted to do it for Engine too, but it is used in the `FlashNode`... I wanted to have a look, but didn't consider duplicating it.

**Pull request merged**